proto/cacheclient.proto

```diff
@@ -403,6 +404,10 @@ message _ListRemoveResponse {}
 
 message _ListFetchRequest {
   bytes list_name = 1;
+  // Inclusive. 0 (start of list) by default
+  sint32 start_index = 2;
+  // If unset or 0, this effectively means list.length()
+  uint32 count = 3;
```

Comment:
Two more things that might be worth noting here.

1. A negative index counts from the end of the list.
2. If count is > the number of elements to return, return as much as you can.